CONTRIBUTING.md

```diff
@@ -1,50 +1,103 @@
 # Contributing to the Project
 
-Thank you for your interest in contributing to the TREND Watermarker!
+Thank you for your interest in contributing to the TREND project!
 
 ## Table of contents
 
 * [CLA](#cla)
-* [Issues and Pull Requests](#issues-and-pull-requests)
-* [License Header](#license-header)
-* [Open Project in IntelliJ](#open-project-in-intellij)
+* [Git(Hub) Process](#github-process)
+    * [Issues](#issues)
+    * [Forks & Branches](#forks--branches)
+    * [Pull Requests](#pull-requests)
+    * [Commits](#commits)
+    * [License Header](#license-header)
+* [IntelliJ](#intellij)
 * [Style Guide](#style-guide)
     * [IntelliJ IDEA configuration](#intellij-idea-configuration)
-    * [Git](#Git)
 
 ## CLA
 
 When contributing to this project, the _Corporate Contributor License Agreement (“CLA”)_ must be
 accepted. See the `CLA.md` file for more information.
 
-## Issues and Pull Requests
+## Git(Hub) Process
+
+Please carefully follow our guidelines for the process with Git and GitHub. Failure to comply 
+may result in rejection of the contribution.
+
+### Issues
 
 To organize and structure all contributions, open an issue for enhancements, feature requests or
-bugs. In case of security issues, please see `SECURITY.md` for more details.
+bugs. Please use the provided templates and fill out all possible sections. In case of security
+issues, see `SECURITY.md` for more details.
+
+### Forks & Branches
+
+When working on an issue, create a branch in the repository (if you have write permission) or fork
+the repository (if you haven't write permission). Create a branch with a descriptive name (like
+`docs/update-readme`), commit your changes as described here and create a pull request afterward.
+
+### Pull Requests
+
+Code changes are handled entirely over pull requests. When proposing a change, create a pull
+request from your working branch to the upstream `main` branch, fill out the template, link it 
+to at least one issue (mandatory) and accept the CLA by keeping the text untouched.
 
-Code changes are handled entirely over pull requests. We use
-the [squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
+We use the
+[squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
 strategy. Therefore, remember to use a pull request title in the
 [conventional commits](https://www.conventionalcommits.org/) format since it is used as the squashed
-commit message.
+commit message. Since this repository uses a
+[monolithic repository approach](https://en.wikipedia.org/wiki/Monorepo), the conventional
+commit _scope_ should be set to the affected component, resulting in the following format for
+the pull request title:
+
+```
+<type>(<component>): <descriptive title>
```

Comment:
we could recommend using [commitlint](https://github.com/conventional-changelog/commitlint)